src/main/steps/applicant/statement-of-service/summary/StatementOfServicePostController.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { getPartyDetails, mapDataInSession } from '../../../../../main/steps/tasklistresponse/utils';
+import {
+  APPLICANT_STATEMENT_OF_SERVICE_NEXT,
+  APPLICANT_STATEMENT_OF_SERVICE_SUMMARY,
+} from '../../../../../main/steps/urls';
+import { CosApiClient } from '../../../../app/case/CosApiClient';
+import { CaseEvent } from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { applyParms } from '../../../../steps/common/url-parser';
+import { prepateStatementOfServiceRequest } from '../choose-parties/StatementOfServiceMapper';
+
+import { form as summaryForm } from './content';
+
+@autobind
+export default class StatementOfServicePostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const form = new Form(summaryForm.fields as FormFields);
+    const { ...formData } = form.getParsedBody(req.body);
+    req.session.errors = form.getErrors(formData);
+    if (!req.session.errors) {
+      req.session.errors = [];
+    }
+    if (req.session.errors?.length) {
+      req.session.save(() =>
+        res.redirect(applyParms(APPLICANT_STATEMENT_OF_SERVICE_SUMMARY, { context: req.params.context }))
+      );
+      return;
+    }
+    const { user, userCase } = req.session;
+    const partyDetails = getPartyDetails(userCase, user.id);
+    const client = new CosApiClient(user.accessToken, req.locals.logger);
+    if (partyDetails) {
+      const userData = prepateStatementOfServiceRequest(req);
```

Comment:
You cannot return the whole usercase from prepateStatementOfServiceRequest. Return only the request and extend the request properties to usercase.

Reply:
It is returning the request itself maybe the variable name it is assigned to is confusing.